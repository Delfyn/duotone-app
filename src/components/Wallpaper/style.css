.image-background-wrapper {
  height: 100vh;
}

.image-background {
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  position: fixed;
  background-position: center;
  background-repeat: repeat;
  background-size: cover;
}

.image-preview {
  box-shadow: 0 0 12px 0 rgba(0, 0, 0, 0.1), 0 10px 30px 0 rgba(0, 0, 0, 0.2);
  max-height: 100vh;
  max-width: 100%;
}
