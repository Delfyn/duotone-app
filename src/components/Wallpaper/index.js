/* @flow */

import React from 'react';
import { DuotoneImage } from 'react-duotone';
import './style.css';

type Props = {
  preview: any,
  primary: string,
  secondary: string,
  fileName: string,
  originalFile: any,
};

const Wallpaper = (
  {
    preview = '',
    primary,
    secondary,
    fileName,
    originalFile,
  }: Props,
) => (
  <div className="image-background-wrapper flex items-center">
    <div
      className="image-background z1"
      style={{ backgroundImage: preview ? `url(${preview})` : null }}
    />
    {originalFile
      ? <DuotoneImage
          className="image-preview mx-auto z2"
          alt={fileName}
          primaryColor={primary}
          secondaryColor={secondary}
          src={originalFile}
        />
      : null}
  </div>
);

export default Wallpaper;
