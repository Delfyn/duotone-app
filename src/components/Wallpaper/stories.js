/* @flow */

import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Wallpaper from './index';

storiesOf('Wallpaper', module).add('Simple example', () => (
  <Wallpaper
    preview="https://facebook.github.io/react/img/logo_og.png"
    primary="#D42D48"
    secondary="#243261"
    fileName="my-image.png"
    originalFile="https://facebook.github.io/react/img/logo_og.png"
  />
));
