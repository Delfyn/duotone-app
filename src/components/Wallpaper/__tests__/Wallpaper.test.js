import React from 'react';
import renderer from 'react-test-renderer';
import Wallpaper from '../index';

test('Wallpaper renders correctly', () => {
  const component = renderer.create(
    <Wallpaper fileName="my-image.png" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
