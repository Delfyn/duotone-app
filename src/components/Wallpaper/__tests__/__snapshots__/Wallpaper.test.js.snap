// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallpaper renders correctly 1`] = `
<div
  className="image-background-wrapper flex items-center"
>
  <div
    className="image-background z1"
    style={
      Object {
        "backgroundImage": null,
      }
    }
  />
</div>
`;
